### PR TITLE
Split 30x redirect and url shortening resolution into separate functions.

### PR DESCRIPTION
**Pull request by fake-name**
I'm not sure what your tolerance for kind-of breaking changes are, but this is a slightly larger scale change that splits the sections that resolve URLs through 30x redirects and the section that resolves a URL from a URL shortening service into two separate functions.

The use case here is basically I'd like to be able to resolve through shortened URLs in a web-spider I'm working on, but I don't want the default behaviour of doing a HEAD request for EVERY URL passed if it *doesn't* fit the known shortener URL patterns. 

Basically, I have ~24 threads doing some HTML processing and link extraction, and I really just want to dump every URL I extract from the spidered content through the unshortener, and if the link's aren't explicitly known shorteners, it should just return the original link. Trying to use the old 30x-if-not-known would result in *thousands* of HTTP head requests, and basically clog up my processing pipeline enormously.

Anyways, the original API interface is still present, though it now does a 30x resolution HEAD request for *every* call, even if it also matched one of the known unshortener. That's mostly just because it was easier to implement that way. There's also two new exposed calls (`unshorten_only()`, `unwrap_30x_only()`) which do unshortening, and HEAD resolution respectively.

This wound up requiring some minor changes to the unit tests, primarily as a number of them return different URLs when `30x` head requested.

------

Also, pirated files in your unit test URLs? I am amused.

### CODE REVIEW

**Comment by fake-name** [comment]
Are you dead?

**Comment by jkehler** [comment]
@fake-name I am indeed still alive. Just been really busy with little time for personal projects lately.

I've looked over your PR and the changes look good to me. I can go ahead and merge it after the README is updated with your new API implementation details. Also if you could bump the version and add your changes to the change log. Then I'll merge it and update PyPi.

**Comment by jkehler** [comment]
Was the LinkBucks changes you made working? I just noticed that the unit test for LinkBucks was commented out.

#5

**Comment by fake-name** [comment]
I never did spend the time getting linkbucks working. I did spend a bit reversing their JS, and it's *really* hard to follow, they're using a bunch of different obfuscation approaches, and it's utterly unreadable, even after reformatting.

I'll see about updating the docs tomorrow.

**Comment by jkehler** [comment]
Sounds good. Maybe I'll have a poke around at LinkBucks again once I have spare time. I also know it's been a pain in the arse to get working hence why I disabled support for it.

**Comment by fake-name** [comment]
Ok, I think that should bring it up to date.

WRT linkbucks, the JDownloader code could be helpful for unwrapping their links, though I was not able to get a translated version of their mechanism for unwrapping linkbucks URLs to work (it's in the repo, though).

**Comment by jkehler** [comment]
Thanks for the contribution. I've merged and have updated PyPi with the latest version. Also I've decided to bump the version to 0.3.0 since it has quite a few API changes.

**Comment by fake-name** [comment]
I had thought about bumping the minor version number, but I thought that since none of the changes actually affected the previously documented API, I'd just bump the patch number. I'm kinda fuzzy on the practical implications of semver anyways.

However you'd prefer to do it is fine.
